backend/common/src/main/java/ai/verta/modeldb/common/db/migration/MigrationTools.java

```diff
@@ -0,0 +1,35 @@
+package ai.verta.modeldb.common.db.migration;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.zip.CRC32;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class MigrationTools {
+  public static String generateLockId(String database, String otherLockInfo) {
+    CRC32 crc32 = new CRC32();
+    crc32.update((database + ":" + otherLockInfo).getBytes(StandardCharsets.UTF_8));
+    return String.valueOf(crc32.getValue());
+  }
+
+  public static void lockDatabase(MigrationDatastore datastore) throws MigrationException {
+    int tries = 0;
+    while (tries++ < 15) {
```

Comment:
Maybe make this count a constant instead of a literal?

Reply:
yeah, that's a good idea. honestly, this whole block of code feels super hacky to me, and I don't like it much.

Reply:
I'll pick this up in a later PR, since I've got two follow-on branches to be managed right now. If that's ok.